opentracing/globaltracer.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) The OpenTracing Authors.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+from __future__ import absolute_import
+
+from threading import Lock
+
+from .tracer import Tracer
+
+
+DEFAULT_TRACER = Tracer()
+
+
+class GlobalTracer(Tracer):
```

Comment:
While the problem is clear to me, I'm not sure we need a Proxy class. Can't we simply store a `_TRACER` with/without the Lock (let's chat in the other thread about it) so that when you `get_global_tracer()` you just return the tracer instance? same applies when you `init_global_tracer(tracer)` where you change the `_TRACER` instance.

My reasoning is:
* you should get the same result
* you avoid code duplication that is needed everytime we change something in the underlying `Tracer` implementation (new API, name change, etc...)

This will avoid a singleton class that is not very common in Python. Of course tell me if I'm missing something.

Reply:
I would be fine with this slightly simpler approach as well - in this case, we would need to 'deprecate' the old `opentracing.tracer` usage, and advise users to use `opentracing.get_global_tracer()` instead (under the hood, we could still be using `opentracing.tracer`, for legacy code, etc).

```python
from opentracing import get_global_tracer(), init_global_tracer()

# previously opentracing.tracer
my_lib_init_tracing(get_global_tracer())
```

Would that be fine with you?

Reply:
Yes I think that would work! But of course this is a suggestion. Happy to hear what others think about it!

Reply:
Sgtm, but I'd name them global_tracer() and set_global_tracer()

Reply:
@yurishkuro Sweet ;) Will update the PR!

Reply:
Agreed!